cmds/core/netstat/netstat_linux.go

```diff
@@ -410,8 +410,8 @@ func command(out io.Writer, args []string) *cmd {
 	// Validate info source flags
 	// none or one allowed to be set
 	if !xorFlags(c.route, c.interfaces, c.groups, c.stats, c.iface != "") {
-		fmt.Printf("%s\n", help)
-		return nil
+		printHelp()
+		log.Fatal("aborting due to conflicting options")
```

Comment:
I personally prefer letting the error bubble up as high as possible and only using the `log.Fatal` when really necessary and/or in top level code. This also makes testing easier.

So what speaks against re-writing this in such a way that it either returns an error or checks if the returned value is nil and then calling `log.Fatal` if that check discovers an error?

But maybe that's just my personal preference, feel free to ignore it.

Reply:
Generally I agree, but this code is already super close to the entry point, and `fs.Parse()` one line above would quite in the same manner under the hood if parsing fails (as the flag set is configured with `ExitOnError`. The added code is basically an extension to `fs.Parse()` so it makes sense to me to use `log.Fatal()` here.

However, to handle it exactly like `fs.Parse()`, you could print "aborting due to conflicting options" and then call `os.Exit(2)` like `fs.Parse()` does (https://pkg.go.dev/flag#ErrorHandling) , log.Fatal() would call `os.Exit(1)`. However I don't think it's worth it.

Reply:
resolved in v2